src/main/scala/strawman/collection/immutable/TreeSet.scala

```diff
@@ -113,4 +115,5 @@ object TreeSet extends SortedIterableFactory[TreeSet] {
       case _ => empty[E] ++ it
     }
 
+  def newBuilder[A : Ordering](): Builder[A, TreeSet[A]] = new ArrayBuffer[A].mapResult(sortedFromIterable[A] _)
```

Comment:
Hm, right, the current collections append directly to the immutable `TreeSet`. We should do the same.